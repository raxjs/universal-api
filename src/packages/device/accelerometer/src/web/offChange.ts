/* eslint-disable @typescript-eslint/no-unused-vars */
import { Callback } from '../types';

const offChange = (cb?: Callback) => {
  throw new Error('Uni API: accelerometer不支持');
};
export default offChange;
