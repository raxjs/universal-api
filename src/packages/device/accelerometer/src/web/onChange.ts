/* eslint-disable @typescript-eslint/no-unused-vars */
import { Callback } from '../types';

const onChange = (cb: Callback) => {
  throw new Error('Uni API: accelerometer不支持');
};
export default onChange;
