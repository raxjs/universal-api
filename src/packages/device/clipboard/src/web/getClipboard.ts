import { normalizeGet } from '../common';
import { CONTAINER_NAME } from '@utils/constant';

const getClipboard = normalizeGet(() => {
  throw new Error('Uni API: getClipboard不支持');
}, CONTAINER_NAME.WEB);

export default getClipboard;
