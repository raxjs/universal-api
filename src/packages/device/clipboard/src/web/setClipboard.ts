import { normalizeSet } from '../common';
import { PARAMS } from '../types';
import { CONTAINER_NAME } from '@utils/constant';

const setClipboard = normalizeSet((args: PARAMS) => {
  const { text = '', success = () => {}, fail = () => {}, complete = () => {} } = args || {};
  let textArea;
  const createTextArea = () => {
    textArea = document.createElement('textArea');
    textArea.style.position = 'absolute';
    textArea.style.top = 0;
    textArea.style.opacity = 0;
    textArea.style.zIndex = -1;
    textArea.value = text;
    document.body.appendChild(textArea);
  };

  const selectText = () => {
    textArea.setAttribute('readonly', '');
    textArea.select();
    textArea.setSelectionRange(0, textArea.value.length);
  };

  const copyToClipboard = () => {
    document.execCommand('copy');
    textArea.remove();
  };

  try {
    createTextArea();
    selectText();
    copyToClipboard();
    success(text);
    complete(text);
    return text;
  } catch (error) {
    fail(error);
    complete(text);
  }
}, CONTAINER_NAME.WEB);

export default setClipboard;
