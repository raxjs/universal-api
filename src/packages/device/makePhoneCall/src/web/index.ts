import { Options } from '../types';
import { normalize } from '../common';
import { CONTAINER_NAME } from '@utils/constant';

const makePhoneCall = (options: Options): void => {
  const phoneNumber = options?.phoneNumber;
  const a = document.createElement('a');
  a.href = `tel:${phoneNumber}`;
  a.innerHTML = `tel:${phoneNumber}`;
  a.style.display = 'none';
  document.body.appendChild(a);

  setTimeout(() => {
    a.click();

    // remove
    setTimeout(() => {
      try {
        document.body.removeChild(a);
      } catch (e) {
        // ignore
      } finally {
        options?.success(); // set resolved
      }
    }, 0);
  }, 0);
};

export default normalize(makePhoneCall, CONTAINER_NAME.WEB);
