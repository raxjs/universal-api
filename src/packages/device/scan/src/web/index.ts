export default () => {
  throw new Error('Uni API: scanCode不支持');
};

