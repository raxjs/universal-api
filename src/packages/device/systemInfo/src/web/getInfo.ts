

import { formatSystemInfo, formatSystemInfoAsync } from '../common';
import { AsyncOptions } from '../types';
import { CONTAINER_NAME } from '@utils/constant';

function getPlatform() {
  const { platform } = navigator;
  if (platform) {
    if (/(iphone|ipod|ipad)/.test(navigator.userAgent.toLowerCase())) return 'iOS';
    if (/android/.test(navigator.userAgent.toLowerCase())) return 'Android';
    return platform;
  }
  return '';
}

const getInfoSync = formatSystemInfo(() => {
  return {
    pixelRatio: window.devicePixelRatio,
    screenWidth: window.document.documentElement.clientWidth,
    screenHeight: window.document.documentElement.clientHeight,
    windowWidth: window.innerWidth,
    windowHeight: window.innerHeight,
    language: navigator.language,
    version: navigator.appVersion,
    platform: getPlatform() || '',
  };
});

const getInfo = formatSystemInfoAsync((options?: AsyncOptions) => {
  const { success, fail, complete } = options;
  try {
    const res = getInfoSync();
    success && success(res);
    complete && complete(res);
  } catch (e) {
    fail && fail(e);
    complete && complete(e);
  }
}, CONTAINER_NAME.WEB);


export default getInfo;
