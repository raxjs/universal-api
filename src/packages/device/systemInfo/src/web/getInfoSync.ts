

import { formatSystemInfo } from '../common';

function getPlatform() {
  const { platform } = navigator;
  if (platform) {
    if (/(iphone|ipod|ipad)/.test(navigator.userAgent.toLowerCase())) return 'iOS';
    if (/android/.test(navigator.userAgent.toLowerCase())) return 'Android';
    return platform;
  }
  return '';
}

const getInfoSync = formatSystemInfo(() => {
  return {
    pixelRatio: window.devicePixelRatio,
    screenWidth: window.document.documentElement.clientWidth,
    screenHeight: window.document.documentElement.clientHeight,
    windowWidth: window.innerWidth,
    windowHeight: window.innerHeight,
    language: navigator.language,
    version: navigator.appVersion,
    platform: getPlatform(),
  };
});


export default getInfoSync;
