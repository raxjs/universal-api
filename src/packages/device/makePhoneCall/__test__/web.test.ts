import { testWebAPI } from '@/utils/__test__/util';
import { DOMUtil } from '@/utils/__test__/web';

testWebAPI('makePhoneCall', async () => {
  /* eslint-disable @typescript-eslint/dot-notation */
  const mockClick = jest.fn();

  // mock for `element.click` function, to avoid problem
  window.HTMLElement.prototype['click'] = mockClick;

  const { default: makePhoneCall } = require('../src/index');
  const p = makePhoneCall({
    phoneNumber: '123',
  });

  const util = new DOMUtil('a');

  expect(util.getComputedStyle().display).toBe('none');
  expect(util.element.getAttribute('href')).toBe('tel:123');
  expect(document.body).toMatchSnapshot();

  await p;
  expect(mockClick.mock.calls.length).toBe(1);
  expect(document.body).toMatchSnapshot();

  /* eslint-enable @typescript-eslint/dot-notation */
});
