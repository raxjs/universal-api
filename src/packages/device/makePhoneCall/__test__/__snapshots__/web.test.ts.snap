// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Platform API: makePhoneCall Test container: web 1`] = `
<body>
  <a
    href="tel:123"
    style="display: none;"
  >
    tel:123
  </a>
</body>
`;

exports[`Test Platform API: makePhoneCall Test container: web 2`] = `<body />`;
